Realm/ObjectStore/shared_realm.hpp

```diff
@@ -158,6 +158,8 @@ namespace realm {
 
         // FIXME private
         Group *read_group();
+        
+        SharedGroup& shared_group();
```

Comment:
@JadenGeller note @tgoyne comment, so the handover logic really should be moved to the object store so that we don't make `SharedGroup` public from binding perspective.